server/src/com/mirth/connect/server/userutil/DestinationSet.java

```diff
@@ -77,15 +82,35 @@ public boolean remove(Object metaDataIdOrConnectorName) {
      *         from processing for this message.
      */
     public boolean remove(Collection<Object> metaDataIdOrConnectorNames) {
-        boolean removed = false;
+        if(metaDataIdOrConnectorNames == null) { return false; }
 
-        for (Object metaDataIdOrConnectorName : metaDataIdOrConnectorNames) {
-            if (remove(metaDataIdOrConnectorName)) {
-                removed = true;
-            }
-        }
+        return metaDataIdOrConnectorNames.stream()
+            .map(this::convertToMetaDataId)
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .map(metaDataIds::remove)
+            .filter(Boolean::booleanValue)
+            .count() > 0;
+
+        // boolean removed = false;
```

Comment:
Dont comment out large blocks of code, just delete it. `git` handles this history if we ever need it